server/src/main/java/org/opensearch/identity/StringPrincipal.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity;
+
+import java.security.Principal;
+import java.util.Objects;
+
+/**
+ * Create a principal from a string
+ */
+class StringPrincipal implements Principal {
```

Comment:
I see that Shiro has 2 different classes that `implement Principal` (within Shiro) called `UsernamePrincipal` and `UserIdPrincipal`. See https://github.com/apache/shiro/tree/df81077726b407f905ba16a9f57ba731b7736375/core/src/test/java/org/apache/shiro/realm

Since `username` is used pretty widely used in OpenSearch as a user identifier should we create a `UsernamePrincipal`?

Reply:
`User` implies there is a _person_ behind the scenes, keeping the usage generic will prevent assumptions about what these principals represent, since it could be an internal system, an extension, or a plugin.